nimble/controller/src/ble_ll_conn.c

```diff
@@ -1808,13 +1810,33 @@ ble_ll_conn_end(struct ble_ll_conn_sm *connsm, uint8_t ble_err)
 {
     struct os_mbuf *m;
     struct os_mbuf_pkthdr *pkthdr;
+    os_sr_t sr;
 #if MYNEWT_VAL(BLE_LL_STRICT_CONN_SCHEDULING)
     os_sr_t sr;
 #endif
 
     /* Remove scheduler events just in case */
     ble_ll_sched_rmv_elem(&connsm->conn_sch);
 
+    /* In case of the supervision timeout we shall make sure
+     * that there is no ongoing connection event. It could happen
+     * because we scheduled connection event before checking connection timeout.
+     * If connection event managed to start, let us drop it.
+     */
+    OS_ENTER_CRITICAL(sr);
+    if (g_ble_ll_conn_cur_sm == connsm) {
+        /* Disable the PHY */
+        ble_phy_disable();
+        /* Disable the wfr timer */
+        ble_ll_wfr_disable();
+        /* Link-layer is in standby state now */
+        ble_ll_state_set(BLE_LL_STATE_STANDBY);
+        /* Set current LL connection to NULL */
```

Comment:
all these comments are redundant since all calls are self-explanatory

and seems like you just copied this from `ble_ll_conn_current_sm_over` so why not make it a common helper?

Reply:
can do it for you

Reply:
@andrzej-kaczmarek ping